### PR TITLE
Login with Google

### PR DESCRIPTION
**Pull request by antoinechalifour**
- Users can now login using their Google accounts
- If they already exist then they just get back an access token and a refresh token
- If they do not exist then they are created on the fly

-> I think there is a security issue with the current implementation : any app can use our `/authorize/google` endpoint to login. Im not sure this is problematic, but in order to fix it we need to move the authorization code fetching to the front end.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -7,4 +7,10 @@ module.exports = {
       },
     },
   },
+  rules: {
+    '@typescript-eslint/camelcase': 'off',
+    'new-cap': 'off',
+    'import/no-unresolved': 'off', // False negatives
+    'import/named': 'off', // False negatives
+  },
```
Comment (antoinechalifour) [inline .eslintrc.js:15]: ESLint rules are 🤷‍♂ and some of them don't make sense in the project.

File: src/domain/usecase/link-social-account.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import * as uuid from 'node-uuid'
+
+import { UserRepository } from '../../application/repository'
+import { User } from '../model'
+
+interface Dependencies {
+  userRepository: UserRepository
+}
+
+export type LinkSocialAccount = (options: {
```
Comment (antoinechalifour) [inline src/domain/usecase/link-social-account.ts:10]: If you guys can suggest a better name for this use case that would be great. Its adding users if they do not exist in the DB after social logins

File: .eslintrc.js [hunk 3]
```diff
@@ -7,4 +7,10 @@ module.exports = {
       },
     },
   },
+  rules: {
+    '@typescript-eslint/camelcase': 'off',
```
Comment (francoischalifour) [inline .eslintrc.js:11]: Is that needed for the `Koa` type?

File: src/domain/usecase/link-social-account.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { UserRepository } from '../..//application/repository'
```
Comment (francoischalifour) [inline src/domain/usecase/link-social-account.test.ts:1]: There's an unwanted additional slash:

```js
import { UserRepository } from '../../application/repository'
```

Was this the cause of the false negative you had with the `import/no-unresolved` rule?

File: .eslintrc.js [hunk 5]
```diff
@@ -7,4 +7,10 @@ module.exports = {
...
+    'new-cap': 'off',
+    'import/no-unresolved': 'off', // False negatives
+    'import/named': 'off', // False negatives
```
Comment (francoischalifour) [inline .eslintrc.js:14]: Why did you need to disable this rule?

File: src/domain/usecase/link-social-account.ts (hunk 2 above)
Comment (francoischalifour) [reply src/domain/usecase/link-social-account.ts:10]: That's a fine name to me.

File: .eslintrc.js (hunk 3 above)
Comment (antoinechalifour) [reply .eslintrc.js:11]: Nope, for third party API response like `access_token`

File: .eslintrc.js (hunk 5 above)
Comment (antoinechalifour) [reply .eslintrc.js:14]: Looks like eslint cannot resolve types definitions for several modules
